lwallet/Command.hs

```diff
@@ -156,6 +163,33 @@ proposeUpdate =
     lexeme parseSoftwareVersion <*>
     many1 parseProposeUpdateSystem
 
+stakeholderIdP :: Parser StakeholderId
+stakeholderIdP = hash
+
+coinPortionP :: Parser CoinPortion
+coinPortionP = do
+    (token, modifier) <- anyText <&> \s -> case Text.stripSuffix "%" s of
+        Nothing -> (s, identity)
+        Just s' -> (s', (/100))
+    case readMaybe (toString token) of
+        Just (modifier -> a) | a >= 0, a <= 1 -> return $ unsafeCoinPortionFromDouble a
+        _                    -> fail "Expected a coin portion"
+
+addrStakeDistrP :: Parser AddrStakeDistribution
+addrStakeDistrP =
+    BootstrapEraDistr <$ text "boot" <|>
+    multiKeyDistrP
+  where
+    multiKeyDistrP = do
+        parts <- many1 ((,) <$> stakeholderIdP <* text ":" <*> coinPortionP)
+        case parts of
+            [(sId, coinPortion)] | coinPortion == unsafeCoinPortionFromDouble 1 ->
```

Comment:
`unsafeCoinPortionFromDouble 1` is `maxBound`, so it might be better to use it here to avoid a partial function.